### PR TITLE
feat: add PeekAll function

### PR DESCRIPTION
**Pull request by li-jin-gou**
#### What type of PR is this?
feat

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.

添加 PeekAll 函数

### CODE REVIEW

**Review by Duslia — DISMISSED** [review]

File: pkg/protocol/header.go [hunk 1]
```diff
@@ -687,6 +690,94 @@ func (h *ResponseHeader) peek(key []byte) []byte {
 	}
 }
 
+// PeekAll returns all header value for the given key.
+//
+// The returned value is valid until the request is released,
+// either though ReleaseResponse or your request handler returning.
+// Any future calls to the Peek* will modify the returned value.
+// Do not store references to returned value. Make copies instead.
```
Comment (welkeyever) [inline pkg/protocol/header.go:698]: 感觉又会有一波数据不一致的oncall出现🤓

**Review by FGYFFFF — DISMISSED** [review]

File: pkg/protocol/header.go (hunk 1 above)
Comment (welkeyever) [reply pkg/protocol/header.go:698]: Also add a `GetAll` for long lifetime & concurrent safety ? 
Corresponds to https://github.com/cloudwego/hertz/blob/develop/pkg/protocol/header.go#L1477

Comment (li-jin-gou) [reply pkg/protocol/header.go:698]: Not for the moment

Comment (li-jin-gou) [reply pkg/protocol/header.go:698]: done

Comment (welkeyever) [inline pkg/protocol/header.go:698]: Add some comments that link to `GetAll`

File: pkg/protocol/header.go [hunk 2]
```diff
@@ -687,6 +690,94 @@ func (h *ResponseHeader) peek(key []byte) []byte {
...
+	}
+	return h.mulHeader
+}
+
+// PeekAll returns all header value for the given key.
+//
+// The returned value is valid until the request is released,
+// either though ReleaseRequest or your request handler returning.
+// Any future calls to the Peek* will modify the returned value.
+// Do not store references to returned value. Make copies instead.
```
Comment (welkeyever) [inline pkg/protocol/header.go:740]: the same here

Comment (li-jin-gou) [reply pkg/protocol/header.go]: done

File: pkg/protocol/header.go (hunk 1 above)
Comment (li-jin-gou) [reply pkg/protocol/header.go]: done

File: pkg/protocol/header.go [hunk 3]
```diff
@@ -687,6 +690,94 @@ func (h *ResponseHeader) peek(key []byte) []byte {
 	}
 }
 
+// PeekAll returns all header value for the given key.
+//
+// The returned value is valid until the request is released,
+// either though ReleaseResponse or your request handler returning.
+// Any future calls to the Peek* will modify the returned value.
+// Do not store references to returned value. Make ResponseHeader.GetAll instead.
```
Comment (welkeyever) [inline pkg/protocol/header.go:698]: Make -> Use;
ResponseHeader.GetAll -> ResponseHeader.GetAll(key)

Comment (li-jin-gou) [reply pkg/protocol/header.go]: done

**Review by welkeyever — APPROVED** [review]
LGTM

**Review by wzekin — APPROVED** [review]

**Pull request merged**